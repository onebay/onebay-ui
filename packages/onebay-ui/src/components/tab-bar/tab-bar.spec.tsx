import { mount } from '@vue/test-utils'
import TabBar from './index'

const factory = (values = {}, slots = { default: [''] }) => {
  return mount(TabBar, {
    slots,
    props: { ...values },
  })
}

const tabList = [
  { title: 'bullet-lis', iconType: 'bullet-list', text: 'new' },
  { title: 'camera', iconType: 'camera' },
  { title: 'folder', iconType: 'folder', text: '100', max: '99' },
  {
    title: 'shopping',
    image:
      'https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png',
    selectedImage:
      'https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png',
    text: 'new',
  },
  {
    title: 'information',
    image:
      'https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png',
    text: 'new',
  },
  {
    title: 'message',
    dot: true,
    text: '30',
    max: 9
  },
]
describe('TabBar Snap', () => {

  it('render initial TabBar', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render TabBar -- props customStyle', () => {
    const wrapper = factory({ customStyle: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render TabBar -- props className', () => {
    const wrapper = factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render TabBar -- props backgroundColor', () => {
    const wrapper = factory({ fixed: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render TabBar -- props tabList', () => {
    const wrapper = factory({ tabList: tabList })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render TabBar -- props iconSize', () => {
    const wrapper = factory({ iconSize: '26', tabList: tabList })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render TabBar -- props fontSize', () => {
    const wrapper = factory({ fontSize: '26', tabList: tabList })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render TabBar -- props color', () => {
    const wrapper = factory({ color: 'red', tabList: tabList })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render TabBar -- props selectedColor', () => {
    const wrapper = factory({ selectedColor: 'red', tabList: tabList })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('render TabBar -- props scroll', () => {
    const wrapper = factory({ scroll: true, tabList: tabList })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('TabBar Event', () => {
  it('TabBar onClick', () => {
    const onClick = jest.fn()
    const wrapper = factory({ onClick: onClick, tabList: tabList })
    wrapper.find('.ob-tab-bar__item').trigger('click')
    expect(onClick).toBeCalled()
  })
})