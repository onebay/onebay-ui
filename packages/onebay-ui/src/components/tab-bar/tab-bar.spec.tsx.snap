// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar Snap render TabBar -- props backgroundColor 1`] = `
<div
  class="ob-tab-bar ob-tab-bar--fixed"
>
  
  
</div>
`;

exports[`TabBar Snap render TabBar -- props className 1`] = `
<div
  class="ob-tab-bar test"
>
  
  
</div>
`;

exports[`TabBar Snap render TabBar -- props color 1`] = `
<div
  class="ob-tab-bar"
>
  
  <div
    class="ob-tab-bar__item ob-tab-bar__item--active"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-bullet-list"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          bullet-lis
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
    style="color: red;"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-camera"
          style="color: red;"
        />
      </div>
      
      <!---->
      
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          camera
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
    style="color: red;"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-folder"
          style="color: red;"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          99+
        </div>
      </div>
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          folder
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
    style="color: red;"
  >
    <!---->
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <img
          class="ob-tab-bar__inner-img ob-tab-bar__inner-img--inactive"
          src="https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png"
        />
        <img
          class="ob-tab-bar__inner-img"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          shopping
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
    style="color: red;"
  >
    <!---->
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <img
          class="ob-tab-bar__inner-img ob-tab-bar__inner-img--inactive"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
        <img
          class="ob-tab-bar__inner-img"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          information
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
    style="color: red;"
  >
    <!---->
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          message
        </div>
        
        <div
          class="ob-badge__dot"
        />
        <!---->
      </div>
    </div>
  </div>
  
</div>
`;

exports[`TabBar Snap render TabBar -- props customStyle 1`] = `
<div
  class="ob-tab-bar"
  style=""
>
  
  
</div>
`;

exports[`TabBar Snap render TabBar -- props fontSize 1`] = `
<div
  class="ob-tab-bar"
>
  
  <div
    class="ob-tab-bar__item ob-tab-bar__item--active"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-bullet-list"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
          style="font-size: 26px;"
        >
          bullet-lis
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-camera"
        />
      </div>
      
      <!---->
      
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
          style="font-size: 26px;"
        >
          camera
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-folder"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          99+
        </div>
      </div>
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
          style="font-size: 26px;"
        >
          folder
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <img
          class="ob-tab-bar__inner-img ob-tab-bar__inner-img--inactive"
          src="https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png"
        />
        <img
          class="ob-tab-bar__inner-img"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
          style="font-size: 26px;"
        >
          shopping
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <img
          class="ob-tab-bar__inner-img ob-tab-bar__inner-img--inactive"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
        <img
          class="ob-tab-bar__inner-img"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
          style="font-size: 26px;"
        >
          information
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
          style="font-size: 26px;"
        >
          message
        </div>
        
        <div
          class="ob-badge__dot"
        />
        <!---->
      </div>
    </div>
  </div>
  
</div>
`;

exports[`TabBar Snap render TabBar -- props iconSize 1`] = `
<div
  class="ob-tab-bar"
>
  
  <div
    class="ob-tab-bar__item ob-tab-bar__item--active"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-bullet-list"
          style="font-size: 26px;"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          bullet-lis
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-camera"
          style="font-size: 26px;"
        />
      </div>
      
      <!---->
      
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          camera
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-folder"
          style="font-size: 26px;"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          99+
        </div>
      </div>
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          folder
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <img
          class="ob-tab-bar__inner-img ob-tab-bar__inner-img--inactive"
          src="https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png"
          style="width: 26px; height: 26px;"
        />
        <img
          class="ob-tab-bar__inner-img"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
          style="width: 26px; height: 26px;"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          shopping
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <img
          class="ob-tab-bar__inner-img ob-tab-bar__inner-img--inactive"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
          style="width: 26px; height: 26px;"
        />
        <img
          class="ob-tab-bar__inner-img"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
          style="width: 26px; height: 26px;"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          information
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          message
        </div>
        
        <div
          class="ob-badge__dot"
        />
        <!---->
      </div>
    </div>
  </div>
  
</div>
`;

exports[`TabBar Snap render TabBar -- props scroll 1`] = `
<div
  class="ob-tab-bar"
  scroll="true"
>
  
  <div
    class="ob-tab-bar__item ob-tab-bar__item--active"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-bullet-list"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          bullet-lis
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-camera"
        />
      </div>
      
      <!---->
      
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          camera
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-folder"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          99+
        </div>
      </div>
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          folder
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <img
          class="ob-tab-bar__inner-img ob-tab-bar__inner-img--inactive"
          src="https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png"
        />
        <img
          class="ob-tab-bar__inner-img"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          shopping
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <img
          class="ob-tab-bar__inner-img ob-tab-bar__inner-img--inactive"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
        <img
          class="ob-tab-bar__inner-img"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          information
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          message
        </div>
        
        <div
          class="ob-badge__dot"
        />
        <!---->
      </div>
    </div>
  </div>
  
</div>
`;

exports[`TabBar Snap render TabBar -- props selectedColor 1`] = `
<div
  class="ob-tab-bar"
>
  
  <div
    class="ob-tab-bar__item ob-tab-bar__item--active"
    style="color: red;"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-bullet-list"
          style="color: red;"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          bullet-lis
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-camera"
        />
      </div>
      
      <!---->
      
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          camera
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-folder"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          99+
        </div>
      </div>
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          folder
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <img
          class="ob-tab-bar__inner-img ob-tab-bar__inner-img--inactive"
          src="https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png"
        />
        <img
          class="ob-tab-bar__inner-img"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          shopping
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <img
          class="ob-tab-bar__inner-img ob-tab-bar__inner-img--inactive"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
        <img
          class="ob-tab-bar__inner-img"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          information
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          message
        </div>
        
        <div
          class="ob-badge__dot"
        />
        <!---->
      </div>
    </div>
  </div>
  
</div>
`;

exports[`TabBar Snap render TabBar -- props tabList 1`] = `
<div
  class="ob-tab-bar"
>
  
  <div
    class="ob-tab-bar__item ob-tab-bar__item--active"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-bullet-list"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          bullet-lis
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-camera"
        />
      </div>
      
      <!---->
      
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          camera
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <div
          class="ob-icon ob-icon-folder"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          99+
        </div>
      </div>
    </div>
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          folder
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <img
          class="ob-tab-bar__inner-img ob-tab-bar__inner-img--inactive"
          src="https://img14.360buyimg.com/jdphoto/s72x72_jfs/t17251/336/1311038817/3177/72595a07/5ac44618Na1db7b09.png"
        />
        <img
          class="ob-tab-bar__inner-img"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          shopping
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <div
      class="ob-badge"
    >
      
      <div
        class="ob-tab-bar__icon"
      >
        <img
          class="ob-tab-bar__inner-img ob-tab-bar__inner-img--inactive"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
        <img
          class="ob-tab-bar__inner-img"
          src="https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png"
        />
      </div>
      
      <!---->
      <div>
        <div
          class="ob-badge__num"
        >
          new
        </div>
      </div>
    </div>
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          information
        </div>
        
        <!---->
        
      </div>
    </div>
  </div>
  <div
    class="ob-tab-bar__item"
  >
    <!---->
    <!---->
    <div>
      <div
        class="ob-badge"
      >
        
        <div
          class="ob-tab-bar__title"
        >
          message
        </div>
        
        <div
          class="ob-badge__dot"
        />
        <!---->
      </div>
    </div>
  </div>
  
</div>
`;

exports[`TabBar Snap render initial TabBar 1`] = `
<div
  class="ob-tab-bar"
>
  
  
</div>
`;
